docs/content/v0.20/release.md

```diff
@@ -0,0 +1,60 @@
+- A git tag is immutable. Once associated with a given Git commit, that association is not changed later.
+- We do not put self-references into Git. For example, making release `1.2.3` doe not require changing any file in Git to have the string `1.2.3` in it.
+
+We have the following limitations.
+
+- The only way to publish artifacts (broadly construed, not (necessarily) GitHub "release artifacts") is to make a release.
+- The only way to test published artifacts is to make a release and test it.
+- Thus, it is necessary to keep users clearly appraised of the quality (or status of evaluating the quality) of each release.
+- Because of the lack of self references, most instructions and tests do not have concrete release identifiers in them; instead, the user has to chose and supply the release identifier. There can also be documentation of a specific past release (e.g., the latest stable release) that uses the literal identifier for that past release.
+- Because of the prohibition of self references, Git will not contain the exact bytes of our Helm chart definitions. Where a Helm chart states its own version or supplies a default value for a container image reference to an image built from the same release, the bytes in Git have a placeholder for that image's tag and the process of creating the published release artifacts fills in that placeholder. Think of this as being analogous to the linking done when building a binary executable file.
+
+## Technology
+
+There will be a GitHub workflow that creates the published artifacts for each Git tag whose name starts with "v". The rest of the tag name is required to be a semver release identifier. For each tag `v$version` the following published artifacts will be created.
+
+- The container image for the kubestellar-controller-manager (KCM), at `ghcr.io/kubestellar/kubestellar-controller-manager` (or should it be `ghcr.io/kubestellar/kubestellar/controller-manager` from the ks/ks repo? or just `ghcr.io/kubestellar/controller-manager`?). Image tag will be `$version`.
+- The Helm chart (for installing the KCM for a WDS), at `ghcr.io/kubestellar/kubestellar-controller-manager-chart` with version `$version` and Helm "appVersion" `$version`. The `values.yaml` of this chart version has a default value for the image reference to the KCM and that value is `ghcr.io/kubestellar/kubestellar-controller-manager:$version`.
+
+## Testing and Instructions
+
+The unit tests (of which we have almost none right now), integration tests (of which we have none yet), and end-to-end (E2E) tests in this repository are run in the context of a local copy of this repository and test that version of this repository --- not using any published release artifacts.
+
+The end-to-end tests include ones written in `bash`, and these are the only documentation telling a user how to use the present version of this repository. Again, these tests do not use any published artifacts from a release of this repo.
```

Comment:
This is a point in time, right ? And we have also several examples, although we make the disclaimer that they are specific to a particular version and not maintained beyond that.

Reply:
I think this section will need to be updated in the next few weeks.  We have an outstanding e2e tests PR that is not based on bash.

Reply:
just as FYI We also plan to have tests that will use artifacts instead of building the code on the fly , but that's indeed a future statement.

Reply:
@ezrasilvera : I addressed that in what I called "release tests".

@pdettori : you effectively asked about `examples.md` above in two other comments. My point in this part of the draft text is that we need a way to tell contributors how to exercise the contents of `main`. I specifically want `bash` scripts because contributors can read them and they are very close to what a contributor would do. If these tests were written in `go` it would be much harder for a contributor to translate that to the shell commands to execute. I am trying to not duplicate material here; duplication would introduce a requirement for checking consistency.

Reply:
I also commented in https://github.com/kubestellar/kubestellar/issues/1612#issuecomment-1944302541